src/web_ui/__init__.py

```diff
@@ -78,6 +78,19 @@ def env_scan():
     )
 
 
+@app.route("/configuration")
+def configuration():
+    device_configs = []
+    config = kvs.get(keys.CONFIG)
+    if config.get('devices'):
```

Comment:
@tienthai-ammp I don't see a reason to use a construct like
```
device_configs = []
...
if config.get('devices'):
    device_configs = config.get('devices')
```
here. Firstly, `device_configs` should be a dict so it doesn't make sense to initialize as a list. Secondly, I often see checks like `if config.get('devices')` being used incorrectly, and I think this is potentially one such case. If we want to check whether a `'devices'` key exists in `config` then the check should be `if 'devices' in config`. The actual check we're doing (a) actually gets the value of the `devices` key, and (b) checks if it's "truthy". This may be ok here but can easily lead to unintended behavior in many cases.

Thirdly, if we're going to do `device_configs = config.get('devices')` anyway, then there's no real point in checking whether the `config['devices']` key exists.

In summary, I think we can just do
```
device_configs = config.get('devices', {})
```
which would simply return an empty dict if the `'devices'` key is missing.

Reply:
Yeah this makes total sense. I was looping through .items() before and append to the list but then changed to this and forgot the different data structure, hence the initialised of `device_config` stays as array.